### PR TITLE
Social Media Website - Milstone #2: friendship v1 

### PR DESCRIPTION
**Pull request by diazgio**
### Social Media Website

in this PR we:
- Add friendship model and controller.
- Add helper methods for my user to add friends and deleted them.
- Add change the user show page to render your new friends and pending friends.
- Add basic OOP.

### CODE REVIEW

File: app/views/users/show.html.erb
Comment (teekaytech) [inline app/views/users/show.html.erb]: All View-related logic can be encapsulated in View helpers (app/helpers/friendship_helper.rb).
You can check [this guide](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) to understand how to move your view-related logic into their corresponding helpers.

File: app/views/users/show.html.erb
Comment (teekaytech) [inline app/views/users/show.html.erb]: This is logic too. Move it to the helper file.

File: app/views/users/show.html.erb
Comment (teekaytech) [inline app/views/users/show.html.erb]: Here's another logic. It should be moved to the helper file.

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made laudable progress in this milestone.
However, some changes are required before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [ ] **There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.** [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)
- Check my comments below to identify the logic wrapped in your view and fix as appropriate.

- [ ] **If user1 and user2 are already friends (i.e. user1 have accepted user2 friend request or vice versa), then on the `ALL USERS` page, we should not have the `ADD FRIEND` button for these 2 users.** 
- Since they are already friends, the `ADD FRIEND` button should not show for either `user` on the `ALL USERS` page.

**The screenshot below shows that `user1` and `user2` are already friends**


**The second screenshot below shows that the `ADD FRIEND` button still shows, this should be removed for friends.**


- [ ] **There should be a button to invite a user to friendship on single user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))**
- Let's say user1 wants to add user2 as a friend, If user1 click on userr2's name to view user2's profile, you should add a button to invite user2 right from user2's page.

- [ ] **Posts timeline should only include posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))**
- Right now, a user can see posts from other users with whom they're not friends.
- You should ensure that a logged in user (current_user) can only see posts made by that user and other users who have already accepted the current user's friend request.

- [ ] **For this milestone, you need to add unit and integration tests!** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- You've not added any test cases, either unit or integration, in your project. Kindly do so.
* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by diazgio** [comment]
Thanks for the review all points are fixed :+1:

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [ ] when User 1 sends a friend request to User 2, User 2 can still see the "Add Friend" button on User 1, instead of "Accept" and "Reject" or something like that

- [ ] Due to the issue above, I can't make any friends. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/pending_request.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- [ ] please add tests to your project. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Comment by diazgio** [comment]
I add the method to see the buttons to accept or reject in the user's path and also the user can see it on his own page.
On the user's page, the owner can see the actual friends, the user that is pending to accept or reject his request, and the requests from other users.
Added the spec files to friendship model and User's Views.

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️
### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**
